addons/portal/controllers/portal.py

```diff
@@ -181,10 +181,85 @@ def home(self, **kw):
         values = self._prepare_portal_layout_values()
         return request.render("portal.portal_my_home", values)
 
+    @route(['/my/account/save/<int:address>'], type='http', auth='user', website=True)
+    def save_address(self, address=False, **kw):
+        address_data = {
+            'name': kw.get('name', ''),
+            'email': kw.get('email', ''),
+            'company_name': kw.get('company_name', ''),
+            'country_id': kw.get('country_id', ''),
+            'vat': kw.get('vat', ''),
+            'phone': kw.get('phone', ''),
+            'street': kw.get('street'),
+            'street2': kw.get('street2', ''),
+            'zip': kw.get('zipcode'),
+            'city': kw.get('city', ''),
+            'state_id': kw.get('state_id'),
+            'type': kw.get('type'),
+        }
+        if address:
+            partner = request.env['res.partner'].sudo().browse(address)
```

Comment:
You don't check anything. Are you sure a portal user could not update the address of another partner ? I could update the email of my competitor to make sure he never receive the mail from the Odoo instance...

Reply:
@arj-odoo  oops sorry, :cry:  i'll update the code.

Thank you so much